DataDefinitions/DestinationSystem.cs

```diff
@@ -10,7 +10,7 @@ public class DestinationSystem
 
         public DestinationSystem(DestinationSystem DestinationSystem)
         {
-            this.name = name;
+            this.name = DestinationSystem.name;
```

Comment:
This constructor appears to be entirely unused. Do we need the constructor at all?

If we need it, we should null protect it. 
```cs
             this.name = DestinationSystem?.name; 
```